dbms/src/Storages/KVStore/FFI/JointThreadAllocInfo.cpp

```diff
@@ -27,7 +27,7 @@ namespace DB
 JointThreadInfoJeallocMap::JointThreadInfoJeallocMap()
 {
     monitoring_thread = new std::thread([&]() {
-        setThreadName("ThreadMemoryTracer");
+        setThreadName("ThdMemTrace");
```

Comment:
How does the name length cause the issue in #9043? Is it just a warning or some real race caused by this long name? Or is it a false-positive?

Reply:
The `std::cerr` is not thread safe, and it cause data race between std::cerr output here and that in other loggings cause by `DB::Logger`.

This data race make `/tiflash/gtests_dbms: Command '['/tiflash/gtests_dbms', '--gtest_list_tests']' returned non-zero exit status 66` and TSAN fails to run following commands

Reply:
Two things I'm still not clear:
1. How does this thread name have anything to do with `std::cerr`?
2. How could simply shortening the name avoid the race?

Reply:
The `setThreadName` here is a cross-platform version. It will print message to `std::err` when the thread name is larger than 15 bytes 😂
https://github.com/pingcap/tiflash/blob/de9f362397cd0ea65c5cc747b762e44d2e03cc9c/dbms/src/Common/setThreadName.cpp#L38-L43

Reply:
Got it, thanks.

Maybe we should replace `std::cerr` with a reasonable logger in the future.